### DISCUSSION TITLE
datatagr: Tagging and Validating Data

### DISCUSSION THREAD

**Discussion by chartgerink** [epiverse-trace/epiverse-trace.github.io#221] [Project Ideas]
### Description

This package would abstract out the tagging and validating functionality from **linelist**, in order to be applied in other domains.

Initially, the package aims to only transport the relevant code out of linelist into its own package. Only later may functionality be adjusted based on feedback from its use in both linelist and beyond. The process of transporting out the relevant code may raise some new questions that need immediate addressing, which will be raised when they come up (e.g., how to approach abstracted out validation for unknown variables).

### Typical end-users

Data analysts/wranglers

### Potential contributors

@Bisaloo

### Key collaborators

-

### Inputs

data.frame

### Outputs

datatagr object (similar to a linelist object)

### Imports

Any providing a regular data frame. As this will be upstream from **linelist**, we can regard anything that works with that as valid input for Epiverse specifically.

### Used by

* **linelist**

### Related projects

Not applicable at the time.

### Usage

```bash
x <- dataset %>%
  tibble() %>%
  tag_data(
    tag1 = "tag_var_1",
    tag2 = "tag_var2",
    tag3 = "var3"
  )
```

### Additional comments

Potential interested parties:
* HARMONIZE - https://www.harmonize-tools.org/

Alternative name options:
* **datatagger** - more verbose
* **datata** - [according to Wiktionary](https://en.wiktionary.org/wiki/datata), a tense of the multiple 'dati' in Esperanto, and a bit funnier 😅 (not clearer per se)

**Comment by Bisaloo** [comment]
Related issue in the linelist repo: https://github.com/epiverse-trace/linelist/issues/78.

**Comment by TimTaylor** [comment]
Unsure on overlap but flagging for awareness:

https://cran.r-project.org/web/packages/validate/index.html
and associated vignettes/papers
https://cran.r-project.org/web/packages/validate/vignettes/cookbook.html
https://cran.r-project.org/web/packages/validate/vignettes/JSS_3483.pdf

**Reply by chartgerink** [reply]
Thanks @TimTaylor 🙌

**Comment by chartgerink** [comment]
Package is started over at [epiverse-trace/datatagr](https://github.com/epiverse-trace/datatagr).

For any further discussions, please go there 😊
